src/Courses/TopicSettings/GenericFormInputs.tsx

```diff
@@ -1,133 +1,247 @@
+                inputRef={register()} 
+                label='Max Graded Attempts'
+                type='number'
+                inputProps={{min: -1}}
+            />
+            <ErrorMessage name={name} errors={errors} />
+        </>
+    );
+};
 
 // TODO: Replace with MUI Autocomplete component
-export const ProblemPath = (register: any) => (
-    <TextField 
-        fullWidth 
-        name='webworkQuestionPath' 
-        inputRef={register()} 
-        label='Problem Path'
-    />
-);
-
-export const ProblemWeight = (register: any) => (
-    <TextField 
-        name='weight' 
-        inputRef={register()} 
-        label='Weight'
-        type='number'
-        inputProps={{min: 0}}
-    />
-);
+export const ProblemPath: React.FC<{}> = () => {
+    const { register, errors } = useFormContext();
+    const name = 'webworkQuestionPath';
+
+    return (
+        <>
+            <TextField 
+                fullWidth 
+                name={name} 
+                inputRef={register()} 
+                label='Problem Path'
+            />
+            <ErrorMessage name={name} errors={errors} />
+        </>
+    );
+};
+
+export const ProblemWeight: React.FC<{}> = () => {
+    const { register, errors } = useFormContext();
+    const name = 'weight';
+
+    return (
+        <>
+            <TextField 
+                name={name} 
+                inputRef={register()} 
+                label='Weight'
+                type='number'
+                inputProps={{min: 0}}
+            />
+            <ErrorMessage name={name} errors={errors} />
+        </>
+    );
+};
+
+export const OptionalField: React.FC<{}> = () => {
+    const { control, errors } = useFormContext();
+    const name = 'optional';
+    
+    return (
+        <>
+            <Controller 
+                name={name}
+                control={control}
+                defaultValue={false}
+                render={({ onChange, onBlur, value, name }) => (
+                    <FormControlLabel 
+                        name='optional'
+                        label={'Optional'} 
+                        labelPlacement='end'
+                        control={
+                            <Switch 
+                                onBlur={onBlur}
+                                onChange={e => onChange(e.target.checked)}
+                                color='primary'
+                                checked={value}
+                                value={value}
+                                name={name}
+                            />
+                        }
+                    />
+                )} 
+            />
+            <ErrorMessage name={name} errors={errors} />
+        </>
+    );
+};
+
+const examProblemFieldNamePrefix = 'courseQuestionAssessmentInfo';
+
+export const RandomSeedSet: React.FC<{}> = () => {
+    const { control, errors } = useFormContext();
+    const name = `${examProblemFieldNamePrefix}.randomSeedSet`;
```

Comment:
what if different object structures were using the same form input? This seems like we are breaking down the ui for specific objects

Also if we are going to do this we should add some type checking:
```
const name = `${examProblemFieldNamePrefix}.${nameof<CourseQuestionAssessmentInfo>('randomSeedSet')}`;
```

Right now the rename to version was done on the backend and in these form elements that use strings but not in the model and we're not catching that

Reply:
This seems like a good candidate to address in a new PR.